### DISCUSSION TITLE
附录中如果含有\ref{sec:  }等，目录中此内容显示过于复杂。

### DISCUSSION THREAD

**Discussion by earthmoon** [whutug/whu-thesis#163] [Q&A] [answered]
你好，
在我们的模板中，在`chapter1.tex`中写入此代码：`\section{哈哈哈哈哈}\label{sec:test}`
然后在`appendix.tex`中写入此代码：

> \chapter{数据}
> \section{对应\ref{sec:test}}

运行后查看目录，会发现目录中显示：
![image](https://user-images.githubusercontent.com/45897471/178725778-c6ee7029-1093-42f4-afe5-f77762ce1fc6.png)
而在附录中显示：
![image](https://user-images.githubusercontent.com/45897471/178725862-9eb8cb04-3e3f-4451-9923-96618dd18338.png)
可以发现目录中的显示是过于复杂的，该怎么修正过来呢？

**Comment by sikouhjw** [comment]
复现不了，请确定你用原模板能复现，如果是自己加了什么宏包，请提供信息。

**Reply by earthmoon** [reply]
> 复现不了，请确定你用原模板能复现，如果是自己加了什么宏包，请提供信息。

我用的之前老模板，我先试着看看是不是增加的宏包造成的。谢谢！

**Reply by earthmoon** [reply]
> 复现不了，请确定你用原模板能复现，如果是自己加了什么宏包，请提供信息。

我的宏包都在这，我刚试了相关的，没有找到问题症结。

>  \documentclass[type=doctor,class=design]{whu-thesis}
> \whusetup
>   {
>    %  info               =
>   %    {
>     %    title          = {1},
>   %     title*         = {An Introduction to LaTeX Thesis Template\\of Wuhan University},
>    %     student-number = {xxxxxxxxxxxxx},
>    %     school         = {1},
>   %      author         = {1},
>    %     author*        = {1},
>    %     subject        = {1},
>    %     major          = {1},
>     %    advisor        = {1},
>    %     direction      = {1},
>     %    date           = {2022/05},
>      %   keywords       = {关键词 1 , 关键词 2 , 关键词 3 , 关键词 4 , 一个非常非常，非常非常长——的关键词 5},
>     %    keywords*      = {key word 1 , key word 2 , key word 3 , key word 4 , {and a very very, very very long key word---the key word 5}},
>    %   },
>     style              =
>       {
>         graphics-path  = {{figures/}{data/}},
>       %  list-of-figures,
>      %   list-of-tables,
>         bib-backend    = biblatex,
>       },
>     element            =
>       {
>       %  innovation     = {pages/innovation},
>      %   abstract       = {pages/abstract},
>       %  abstract*      = {pages/enabstract},
>        bibliography   = {ref/refs.bib , ref/thu.bib},
>       %  achievements   = {pages/achievements},
>       %  thanks         = {pages/thanks},
>         appendix       = {pages/appendix}
>       }
>   }
> \usepackage[]{hyperref}%协助cleveref
> \usepackage[version=4]{mhchem} %化学包
> \usepackage{tabularray}%latex3e 强大的表格制作包
> \usepackage{svg}%插入svg图
> \usepackage{xcolor}%字体颜色和背景颜色包
> \usepackage{mathtools}% mathtools builds on and extends amsmath package
> \usepackage{chngpage}%临时调整页面间距，比如公式太长了
> \usepackage[export]{adjustbox}[2011/08/13]%调整图片宽度时允许居中
> \usepackage{lipsum}%临时生成段落
> \usepackage{pdflscape}
> \usepackage{rotating}
> \usepackage{afterpage}
> \usepackage{scrextend}
> \usepackage[customcolors]{hf-tikz}
> \usepackage{tikz}
> \usepackage{scrextend}%用来修改页边距的，使用时是\begin{addmargin}...\end{margin}
> \usepackage{lipsum}
> \usepackage{chemmacros,chemformula}%最新最广泛的化学宏包
> \usetikzlibrary{arrows.meta,decorations.pathmorphing,backgrounds,positioning,fit,petri,intersections,angles,quotes,calc,through,trees,mindmap,shadows}
> 
> \newcommand*{\eqcitet}[2]{%公式中插入文献，可以使用页码，靠右
> 	\vadjust{%
> 		\smallskip
> 		\hbox to \linewidth{\hfill\citet[#1]{#2}}%
> 	}%
> }
> 
> \newenvironment{absolutelynopagebreak}
> {\par\nobreak\vfil\penalty0\vfilneg
> 	\vtop\bgroup}
> {\par\xdef\tpd{\the\prevdepth}\egroup
> 	\prevdepth=\tpd}
> 
> \newcommand{\pagerefw}[1]{~\pageref{#1}~}%使用\pagerefw{}可以使得页码数字两端空格合适
> \hypersetup{bookmarksnumbered=true}%生成pdf时，让目录标签带上数字
> \UseTblrLibrary{booktabs, siunitx}%
> 
> \newcommand{\RNUM}[1]{\uppercase\expandafter{\romannumeral #1\relax}}
> \newcommand{\rnum}[1]{\romannumeral #1\relax}
> \newcommand{\fullref}[1]{\hyperref[#1]{\ref*{#1}\ \nameref*{#1}}}%引用时，既有数字，也有名称，有\fullref{}
> \sisetup{per-mode=symbol}%让单位和数字中的斜杠显示出来
> \def\equationautorefname{式}%\autoref时，将式Equation为式
> \def\figureautorefname{图}%\autoref时，将Figure改为图
> \def\tableautorefname{表}%\autoref时，将Table改为表
> \newlength\labellength
> \DefTblrTemplate{remark}{default}{%
> 	\MapTblrRemarks{%
> 		\settowidth{\labellength}{\UseTblrTemplate{remark-tag}{default}\UseTblrTemplate{remark-sep}{default}}%
> 		\UseTblrTemplate{remark-tag}{default}\UseTblrTemplate{remark-sep}{default}%
> 		\parbox[t]{\dimexpr\textwidth-\labellength}{\UseTblrTemplate{remark-text}{default}}%
> 		\par
> }}%调整表格注释缩进用的
> \DefTblrTemplate{remark}{indentednotes}{%
> 	\MapTblrRemarks{%
> 		\settowidth{\labellength}{\UseTblrTemplate{remark-tag}{default}\UseTblrTemplate{remark-sep}{default}}%
> 		\UseTblrTemplate{remark-tag}{default}\UseTblrTemplate{remark-sep}{default}%
> 		\parbox[t]{\dimexpr\linewidth-\labellength}{\UseTblrTemplate{remark-text}{default}}%
> 		\par}
> }%定义表格缩进的主题
> \NewTblrTheme{indentednotes}{
> 	\SetTblrTemplate{remark}{indentednotes}
> }%定义表格缩进的主题
> \definecolor{tablebackground}{RGB}{245,246,246}
> \SetTblrInner[tblr]{cells={bg=tablebackground}}
> \SetTblrInner[longtblr]{cells={bg=tablebackground}}
> \SetTblrInner[talltblr]{cells={bg=tablebackground}}
> \AtEndPreamble
> {
> 	\hypersetup
> 	{
> 		colorlinks  = true,% 全部超链接都会有颜色
> 		urlcolor    = blue,% 设置 `\url` 的颜色为蓝色
> 		citecolor   = black,%设置正文引用文献颜色为黑色
> 		linkcolor   = black,%设置文内跳转连接为黑色
> 	}
> }
> \BeforeBeginEnvironment{document}
> {
> 	\renewcommand{\bibfont}{\zihao{5}}	
> }
> 
> 
> \begin{document}
> \setmathfont[range=up/{num}]{Times New Roman}
> %%----------- 主体部分 ----------- %%
> \include{pages/chapter1}
> %\include{pages/chapter2}
> %\include{pages/chapter3}
> %\include{pages/chapter4}
> \end{document}

**Comment by earthmoon** [comment]
这一问题百思不得其解：

1. 将最新的cls替换我的cls文件，错误仍旧存在，说明不是cls文件造成；
2. 将whu-thesis-demo.tex替换到最新的武大latex模板里，也没有错，说明不是whu-thesis-demo.tex的问题

那么问题在哪？

**Comment by earthmoon** [comment]
能不能帮忙找下问题，学校最新的whu-thesis-demo.tex和最新的cls文件，可以复现这个问题。这个文档没有什么内容，几乎就是空白，但是出现这个问题，也没有添加任何一个宏包。
[2whu-thesis-0.6e2 (2).zip](https://github.com/whutug/whu-thesis/files/9316680/2whu-thesis-0.6e2.2.zip)

**Comment by earthmoon** [comment, marked as answer]
这个问题很可能是我们模板中cls文件中的bug，修复的办法如下：
把这三行
https://github.com/whutug/whu-thesis/blob/master/whu-thesis.cls#L726-L728
移到
https://github.com/whutug/whu-thesis/blob/master/whu-thesis.cls#L724
之上。
即
![image](https://user-images.githubusercontent.com/45897471/184522773-c2015750-d21f-460f-b05a-d8352de502cb.png)
变为
![image](https://user-images.githubusercontent.com/45897471/184522785-49d89487-55d3-4807-b0eb-fdab24d18499.png)


可以解决
